### PR TITLE
Patch Request added to Web Component

### PR DESCRIPTION
**Pull request by Blueknight10001**
My first pull request for app inventor, I've tried to follow the documentation as well as I can, but it is a tad confusing.

As for the content of the PR itself, this should solve issue #2192  it simply emulates the POST and PUT request methods, adding 3 new block methods to the Web component: PatchText, PatchTextWithEncoding, and PatchFile.

### CODE REVIEW

**Comment by AppInventorWorkerBee** [comment]
Can one of the admins verify this patch?

**Comment by ewpatton** [comment]
@AppInventorWorkerBee ok to test

**Comment by ewpatton** [comment]
Possibly change it to ERROR_WEB_UNABLE_TO_MODIFY_RESOURCE and update the text of the errors in ErrorMessages.java if necessary to match.

File: appinventor/appengine/src/com/google/appinventor/client/youngandroid/YoungAndroidFormUpgrader.java [hunk 1]
```diff
@@ -1618,7 +1618,13 @@ private static int upgradeWebProperties(Map<String, JSONValue> componentProperti
       // of XML using dictionaries.
       srcCompVersion = 7;
     }
+    if (srcCompVersion < 8)  {
+    	// The methods PatchText, PatchTextWithEncoding, and PatchFile were added.
+        // No properties need to be modified to upgrade to version 8.
+        srcCompVersion = 8;
+      }
```
[the remark this answers is not in the document]
Comment (Blueknight10001) [reply appinventor/appengine/src/com/google/appinventor/client/youngandroid/YoungAndroidFormUpgrader.java]: Fixed in latest commit

**Comment by Blueknight10001** [comment]
Is the way I've done the error message changes acceptable?

**Comment by ewpatton** [comment]
We should probably keep both constants around with the same value just in the off chance that an extension relies on the field being available under the old name. The thing that doesn't seem right though is that you are still ending up with a number of existing changes even after the rebase. You may want to rebase interactively and ensure only your changes end up in the branch.

**Comment by Blueknight10001** [comment]
I readded the deleted errors and appended these new ones, but that just feels clunky to me

**Comment by ewpatton** [comment]
@Blueknight10001 Sorry if I confused you. The intention wasn't to add new values for constants, but rather to have new names for the existing constants. For example:

```java
public static final ERROR_WEB_UNABLE_TO_MODIFY_RESOURCE = ERROR_WEB_UNABLE_TO_POST_OR__PUT;
public static final ERROR_WEB_UNABLE_TO_MODIFY_FILE = ERROR_WEB_UNABLE_TO_POST_OR_PUT_FILE;
```

This way the old names remain but the new fields better reflect the meaning of the error (and aren't tied to the HTTP verbs in case we add other methods in the future)

**Comment by Blueknight10001** [comment]
I fixed up all the confusion I think.

**Review by ewpatton — APPROVED** [review]

**Pull request merged**